BaragonServiceIntegrationTests/src/test/java/com/hubspot/baragon/BaragonServiceTestIT.java

```diff
@@ -0,0 +1,478 @@
+     }
+     protected void failed(Throwable e, Description description) {
+       System.out.println("\u001B[31mTest failed\u001B[0m");
+     }
+  };
+  
+  
+  @Rule
+  public Stopwatch stopwatch = new Stopwatch() {
+      @Override
+      protected void finished(long nanos, Description description) {
+        System.out.println("(" + nanos / 1000000000.0 + " s)");
+      }
+  };
+  
+  @After
+  public void teardown() {
+    baragonServiceClient = null;
+  }
+
+  // ----------------------------------------------------
+  
+  @Test
+  public void testStatus() throws Exception {
+    Optional<BaragonServiceStatus> status = baragonServiceClient.getAnyBaragonServiceStatus();
+    assertTrue(status.get().isLeader());
+  }
+
+  @Test
+  public void testWorkers() throws Exception {
+    assertFalse(baragonServiceClient.getBaragonServiceWorkers().isEmpty());
+  }
+  
+  @Test
+  public void testLoadBalancers() throws Exception {
+    assertFalse(baragonServiceClient.getLoadBalancerGroups().isEmpty());
+  }
+  
+  @Test
+  public void testClusterAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
+    assertTrue(metadata.getBaseAgentUri() != null && !metadata.getBaseAgentUri().isEmpty());
+  }
+  
+  @Test
+  public void testClusterKnownAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupKnownAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
+    assertTrue(metadata.getBaseAgentUri() != null && !metadata.getBaseAgentUri().isEmpty());
+    
+    testDeleteKnownAgent();
+  }
+  
+  // ----------------------------------------------------
+  
+  @Test
+  public void testValidRequest() throws Exception {
+    String serviceId = setupTestService();
+    while(baragonServiceClient.getRequest(serviceId).get().getLoadBalancerState().equals(BaragonRequestState.WAITING)) { }
+    
+    if(baragonServiceClient.getRequest(serviceId).get().getLoadBalancerState().equals(BaragonRequestState.SUCCESS)) {
+      testGetService();
+      testState();
+      testClusterBasePaths();
+      
+      removeService(serviceId);
+    } 
+    else {
+      BaragonResponse resp = baragonServiceClient.getRequest(serviceId).get();
+      throw new Exception("Request did not succeed: " + resp.getMessage().get().toString() 
+          + "\nState: " + resp.getLoadBalancerState());
+    }
+  }
+  
+  public void testClusterBasePaths() {
+    String groupName = baragonServiceClient.getLoadBalancerGroups().toArray(new String[0])[0];
+    String[] basePaths = baragonServiceClient.getOccupiedBasePaths(groupName).toArray(new String[0]);
+    assertTrue(basePaths.length > 0);
+    BaragonService service = baragonServiceClient.getServiceForBasePath(LOAD_BALANCER_GROUP, basePaths[0]).get();
+    assertTrue(service.getServiceId() != null);
```

Comment:
The returned service is an optional to start with, if it isn't present, the .get() on the previous line will fail. We can get back the Optional<BaragonService> , assert it is present, and also probably assert that the service name is what we expect it to be for completeness